gutils/gutils.c

```diff
@@ -70,3 +70,23 @@ time_t GetST_MTime(struct stat s) {
 
 	return st_time;
 }
+
+char* FF_HashData(unsigned char* input, int len) {
+    GChecksum* gcs = g_checksum_new(G_CHECKSUM_SHA256);
+    g_checksum_update(gcs, input, len);
+    const char* gcs_cs = g_checksum_get_string(gcs);
+    char* cs = malloc(strlen(gcs_cs)+1);
```

Comment:
Use `copy`. That or don't get the string representation, and use store the raw bytes instead.

Reply:
It's much easier to just use the string representation. I don't really ever need the bytes, and the strings are still comparable. I considered storing the raw bytes instead and converting them and don't think it's worth it.

So I'll use `copy` :smile: